### PR TITLE
ANTs wrapper

### PR DESCRIPTION
**Pull request by piloubazin**
Tested, reformatted.. hopefully all good.

### CODE REVIEW

**Comment by juhuntenburg** [comment]
Hi @piloubazin : can you remove the changes to utils and cruise as well as the build-nodependencies.sh file? I can also do it, let me know. Also, we should probably make an example for this too?

**Comment by piloubazin** [comment]
OK, that should be good now :)

**Comment by juhuntenburg** [comment]
Cool! But we have to find a way to make it only import the ants interface if nipype is installed. Otherwise, when trying to import nighres and not having nipype it gives an import error. I can look into that the next days, should be quick!

**Comment by piloubazin** [comment]
OK, I have made the Nipype dependencies optional and finally tracked down the utils bug with empty output directories. Ready to merge?

**Comment by juhuntenburg** [comment]
Cool, let me try it out once tomorrow ya?

**Comment by piloubazin** [comment]
@juhuntenburg so, all good?

**Comment by juhuntenburg** [comment]
Hi sorry for the delay. No, this still makes Nighres unusable if Nipype is not installed. I would remove the warning and sys.exit() and just pass silently if Nipype is not there. Then, only when trying to use the ants function raise an error if Nipype is not there. Also we should write clearly on top of the docstring that Nipype and ANTs need to be installed and link to the websites (links could be in the notes if too long)

**Comment by piloubazin** [comment]
Could you give it a try, then? I am not sure what you mean by passing silently, sorry..

**Comment by juhuntenburg** [comment]
Yes will do! Might take me a few days to get to it though

File: nighres/registration/embedded_syn.py [hunk 1]
```diff
@@ -0,0 +1,255 @@
+# basic dependencies
+import os
+import sys
+
+# main dependencies: numpy, nibabel
+import numpy as np
+import nibabel as nb
+
+# for external tools: nipype
+
```
Comment (juhuntenburg) [inline nighres/registration/embedded_syn.py]: remove the following lines try /except statement

File: nighres/registration/embedded_syn.py [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+    References
+    ----------
+    .. [1] Avants et al (2008), Symmetric diffeomorphic 
+       image registration with cross-correlation: evaluating automated labeling 
+       of elderly and neurodegenerative brain, Med Image Anal. 12(1):26-41
+    .. [2] Gorgolewski et al (2011) Nipype: a flexible, lightweight and extensible 
+       neuroimaging data processing framework in python. Front Neuroinform 5. 
+       doi:10.3389/fninf.2011.00013
+    """
+
```
Comment (juhuntenburg) [inline nighres/registration/embedded_syn.py]: Include the import here: 

    try:
        from nipype.interfaces.ants import ANTS
        from nipype.interfaces.ants import ApplyTransforms
    except ImportError:
        print('Please install Nipype to use this function')
        return

**Comment by juhuntenburg** [comment]
So this fixes the problem of Nighres not running at all without Nipype. But two remaining issues
1. How do we test ANTS is installed? 
2. Something is still wrong with the output_dir function, trying to figure out what

**Comment by juhuntenburg** [comment]
Hm, not sure. I get this error when trying to run embedded_syn with only source and target image as input: 

```
--> 143     save_volume(os.path.join(output_dir, src_map_file), src_map)
    144     for x in xrange(ntx):
    145         for y in xrange(nty):

/usr/lib/python2.7/posixpath.pyc in join(a, *p)
     68         if b.startswith('/'):
     69             path = b
---> 70         elif path == '' or path.endswith('/'):
     71             path +=  b
     72         else:

AttributeError: 'NoneType' object has no attribute 'endswith'
```

**Comment by juhuntenburg** [comment]
So somehow output_dir is "None" I think. I have no more time now, but I can look into it another time

**Pull request closed without merging**